Shared/Samples/Display route layer/DisplayRouteLayerView.swift

```diff
@@ -0,0 +1,118 @@
+    @State private var directions = [String]()
+    
+    /// A Boolean value indicating whether to show the directions.
+    @State private var isShowingDirections = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapView(map: map)
+            .task {
+                do {
+                    // Create a portal with a portal item using the item ID for route data in Portland, OR.
+                    let portalItem = PortalItem(
+                        portal: .arcGISOnline(connection: .anonymous),
+                        id: .portlandRoute
+                    )
+                    
+                    // Create a collection of features using the item.
+                    let featureCollection = FeatureCollection(item: portalItem)
+                    
+                    // Load the feature collection.
+                    try await loadFeatureCollection(featureCollection)
+                    
+                    // Create a feature collection layer using the feature collection.
+                    let featureCollectionLayer = FeatureCollectionLayer(featureCollection: featureCollection)
+                    
+                    // Add the feature collection layer to the map's operational layers.
+                    map.addOperationalLayer(featureCollectionLayer)
+                } catch {
+                    self.error = error
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button {
+                        isShowingDirections = true
+                    } label: {
+                        Image(systemName: "arrow.triangle.turn.up.right.diamond")
+                    }
+                    .disabled(directions.isEmpty)
+                    .popover(isPresented: $isShowingDirections) {
+                        NavigationStack {
+                            List(directions, id: \.self) { direction in
+                                Text(direction)
+                            }
+                            .navigationTitle("Directions")
+                            .navigationBarTitleDisplayMode(.inline)
+                            .toolbar {
+                                ToolbarItem(placement: .confirmationAction) {
+                                    Button("Done") {
+                                        isShowingDirections = false
+                                    }
+                                }
+                            }
+                        }
+                        .frame(idealWidth: 320, idealHeight: 528)
+                    }
+                }
+            }
+            .errorAlert(presentingError: $error)
+    }
+    
+    /// Loads the feature collection.
+    /// - Parameter featureCollection: The feature collection.
+    func loadFeatureCollection(_ featureCollection: FeatureCollection) async throws {
```

Comment:
It seems like the name of this function should better describe what all is being done in it.